webrender/src/frame_builder.rs

```diff
@@ -1535,9 +1535,10 @@ impl<'a> LayerRectCalculationAndCullingPass<'a> {
             };
 
             if bounding_rect.is_none() {
-                bounding_rect =
-                    Some(clip_info.xf_rect.as_ref().map_or_else(DeviceIntRect::zero,
-                                                                |x| x.bounding_rect))
+                bounding_rect = Some(match clip_info.xf_rect {
```

Comment:
If you want to calculate the intersection sooner then I recommend that you rename the `xf_rect` members to something like `screen_bounding_rect` and update the comments to reflect that they are the intersected onscreen bounding rects for their respective data structures.